### ISSUE TITLE
Connect to repl triggers all midje tests?

### ISSUE THREAD

**Issue by arichiardi** [marick/Midje#319]
Hello, I am observing a weird behaviour using ```midje``` ```2.0.0-SNAPSHOT```.

Basically my workflow is:

- Start a ```lein repl```
- Connect to it from outside (tipically ```cider```'s without jacking-in)
- Work

The thing is, every time I connect to the repl, all the tests are triggered, without even considering my (which work fine with ```lein midje``` instead:
```clojure
(change-defaults :print-level :print-namespaces
                 :fact-filter #(and (not (:slow %1))
                                   (not (:bench %1))
                                   (not (:profile %1))))```

What am I doing wrong? Is this expected?

**Comment by marick** [comment]
Does it happen on 1.7? All that's been done with 2.0 so far is moving stuff out into separate libraries.

**Comment by arichiardi** [comment]
Yes, just tried, and it runs all the test without even filtering...

**Comment by marick** [comment]
So, to make sure I understand: both 1.7 and 2.0 behave the same way: they run all the tests without any change to any file, and they don't obey the `:fact-filter`? 

Since I run `(autotest)` in an emacs shell buffer (not using Cider), I'm not sure how to interpret your problem. You run `lein repl` in an ordinary shell? What do you type after that? How do you connect to the repl? (And how often?)

**Comment by arichiardi** [comment]
I run ```lein repl``` from the shell (and nothing else, on linux) and then connect to it through cider's repl connect. As soon as I see that cider is connected, the repl instance in the shell starts crunching tests, without ```:fact-filtering```. I am sure of it because the filtering works perfectly with ```lein midje``` , i.e. I don't see any heavy test.
Is there a way to debug the issue, I can help you with that.

**Comment by marick** [comment]
You must type something after `lein midje` and before connecting. Autotest? 

Can you make a video capture of what you're doing. (If on a mac, the quicktime player lets you capture your screen.)

**Comment by arichiardi** [comment]
No no my command is lein repl, not lein midje...and then connect to it from cider. .. yes I will record my desktop no problem

On July 10, 2015 9:41:17 PM GMT+02:00, Brian Marick <notifications@github.com> wrote:
>You must type something after `lein midje` and before connecting.
>Autotest? 
>
>Can you make a video capture of what you're doing. (If on a mac, the
>quicktime player lets you capture your screen.)
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/marick/Midje/issues/319#issuecomment-120504556

-ar
